meta-balena-common/recipes-support/resin-init/resin-init-flasher/resin-init-flasher

```diff
@@ -509,6 +512,21 @@ sync
 inform "Board specific flash procedure..."
 /usr/bin/resin-init-flasher-board
 
+EFIPART_LABEL="resin-boot"
+LOADER_PATH="/EFI/BOOT/bootx64.efi"
+BOOT_ENTRY_LABEL="balenaOS"
+if [ "$EFI" = "1" ]; then
+    if [ "$LUKS" = "1" ]; then
+        EFIPART_LABEL=balena-efi
+    fi
+
+    efibootmgr --create \
+               --disk "/dev/$(lsblk -ndo pkname "$(get_dev_path_from_label ${EFIPART_LABEL})")" \
+               --part "$(get_part_number_by_label "$(basename "${internal_dev}")" "${EFIPART_LABEL}")" \
+               --label "${BOOT_ENTRY_LABEL}" \
+               --loader "${LOADER_PATH}"
+fi
+
```

Comment:
@alexgg Correct, this has nothing to do with secure boot, but rather UEFI. The firmware itself does not automatically create entries or change the boot order. It _will_ autodetect bootable EFI binaries with the correct name and path on supported filesystems, and allow those to be booted. If there are no other bootable entries, it's generally true that the autodetected entry will be booted.

The purpose behind creating an explicit entry is to ensure it's booted before other devices, such as the flasher. In the case of QEMU, this ensures that after flashing, the newly installed OS is booted up without removing the installation medium, or manually changing the boot order. This also applies to great deal of other machines which would otherwise continue booting the flasher without additional input.

It's also worth mentioning that this is how installation is already handled in [balena-intel](https://github.com/balena-os/balena-intel/blob/afb28a9df0f8aec9f1ab199b456c9de9a66733b6/layers/meta-balena-genericx86/recipes-support/resin-init/resin-init-flasher-board/resin-init-flasher-board#L66), and this change simply consolidates the approach with balena-generic.

Reply:
Understood @jakogut probably better if they were on a separate PR not to mix things